packages/nimble-components/src/utilities/style/types.ts

```diff
@@ -0,0 +1,5 @@
+export const ZIndexLevels = {
```

Comment:
Part of [the discussion](https://github.com/ni/nimble/pull/530#discussion_r865118893) linked from the issue mentions goals of either naming or documenting the rationale behind the values chosen so that users of the enum can have a better idea of how to choose between the values. Can you take a stab at that or make a case for not doing it?

Reply:
Sorry, I should have given a rationale for not adding documentation. I could not find any z-index value of 200 used by a grid (or anything else) as referenced in that discussion thread. I did not find any other meaningful z-index relationships that we might codify into enum names or document. There isn't any guidance to give users of the enum. They just have to pick a numeric value (which is part of the enum value's name) based on the z-index of whatever other element(s) they're concerned with. However, as your other comment pointed out, it would be worth adding something to our CSS Guidelines about needing to use an enum value, and how to create a new one if necessary.

Reply:
> There isn't any guidance to give users of the enum

So my question is are there z-index values that our apps are using to fight with parts of nimble-components. And that can result in two follow-up questions:
1. What will break with this PRs changes to enum values?
2. Do we need to export enum values that can be used externally? In which case they would be exported as CSS design tokens so users can reliably position things in stacks against our components. If that's the case I don't really know what it would look like, but might need things like tableZIndex and tableZIndexPlus10 or something.

I didn't claim to so a thorough search of how users may be impacted but here is at least one thing that looks fishy:

This is a search for [z-index in Skyline /Web](https://dev.azure.com/ni/_search?action=contents&text=z-index&type=code&lp=custom-Collection&filters=ProjectFilters%7BDevCentral%7DRepositoryFilters%7BSkyline%7DBranchFilters%7Bmaster%7DPathFilters%7B/Web%7D&pageSize=25&result=DefaultCollection/DevCentral/Skyline/GBmaster//Web/Workspaces/SystemLinkShared/projects/systemlink-repository-lib/src/repository/components/store-downloads-table/store-downloads-table.component.scss) and that linked example is for a component using a nimble table that has some kind of wrapper and z-indexes set.

Reply:
> are there z-index values that our apps are using to fight with parts of nimble-components?

It's hard to understand why `z-index` is being set in many of those use cases. I don't see any evidence of them being used to "one-up" any Nimble components with explicit `z-index` values.

> 1. What will break with this PR's changes to enum values?

Not sure I understand. This PR does not change any `z-index` values of any elements. And there are no pre-existing enum values being changed.

> 2. Do we need to export enum values that can be used externally?

I don't think so, given that I don't see (or anticipate) a client need to stack elements in relation to anything in a Nimble component's shadow DOM. If there were such a need, then we would want semantically meaningful tokens, like `popUpMenuZIndex` or `tooltipZIndex` rather than the generic, number-based names (`zIndex1000`) of the enum.

Reply:
> This PR does not change any z-index values of any elements. And there are no pre-existing enum values being changed.

True, doesn't change. Think eventually the dropdown and and anchored region will be replaced by popover api usage so most of z-index usage will be gone at that point.

> I don't see any evidence of them being used to "one-up" any Nimble components with explicit z-index values.

I'll take your evaluation of it. May consider some validation after merging.